src/NUnitFramework/framework/Internal/TestProgressReporter.cs

```diff
@@ -39,19 +43,39 @@ public void TestStarted(ITest test)
             var parent = GetParent(test);
             try
             {
-                string report;
+                var report = _testStartedStringBuilder ??= new StringBuilder();
+                report.Clear();
                 if (test is TestSuite)
                 {
+                    report
+                        .Append("<start-suite id=\"").Append(test.Id).Append('"')
```

Comment:
comment: There was some duplicate of attributes in both paths here, but this move to StringBuilder also probably opens some opportunities for deduplication of XML attribute writing too in the future. Thanks!

Reply:
Fixed in the latest force-push, adding common once before checking the type of requested output